src/preferences/dialog/dlgprefmixer.cpp

```diff
@@ -779,77 +780,84 @@ void DlgPrefMixer::slotUpdate() {
-    constexpr int kGrindYLines = 6;
-
-    int frameWidth = graphicsViewXfader->frameWidth();
-    int sizeX = graphicsViewXfader->width() - 2 * frameWidth;
-    int sizeY = graphicsViewXfader->height() - 2 * frameWidth;
-
-    // Initialize Scene
+    // Initialize or clear scene
     if (m_pxfScene) {
         m_pxfScene->clear();
     } else {
         m_pxfScene = make_parented<QGraphicsScene>(this);
         // The size of the QGraphicsView doesn't change so we need to do this only once
+        graphicsViewXfader->setLineWidth(1); // frame width
+        int sizeX = graphicsViewXfader->width() - 2;
+        int sizeY = graphicsViewXfader->height() - 2;
         m_pxfScene->setSceneRect(0, 0, sizeX, sizeY);
         m_pxfScene->setBackgroundBrush(Qt::black);
+        graphicsViewXfader->setRenderHints(QPainter::Antialiasing);
         graphicsViewXfader->setScene(m_pxfScene);
     }
 
     // Initialize QPens
-    QPen gridPen(Qt::green);
-    QPen graphLinePen(Qt::white);
+    QPen gridPen(Qt::gray);
+    QPen gainPen(Qt::white);
+    QPen totalPen(Qt::red);
+    // In conjunction with anti-aliasing this gives smooth, solid lines.
+    totalPen.setWidth(2);
+    gainPen.setWidth(2);
+    // For some reason grid lines also appear 2px wide, with the nice side effect
+    // that gain curves now intersect 'exactly'at the grid center line.
+
+    const int sceneW = static_cast<int>(m_pxfScene->width());
+    const int sceneH = static_cast<int>(m_pxfScene->height());
 
     // Draw grid
-    for (int i = 1; i < kGrindXLines; i++) {
-        m_pxfScene->addLine(
-                QLineF(0, i * (sizeY / kGrindXLines), sizeX, i * (sizeY / kGrindXLines)), gridPen);
+    const int kGridHLines = 3;
+    const int kGridVLines = 5;
+    const int kGridHDist = static_cast<int>(sceneH / (kGridHLines + 1));
+    const int kGridVDist = static_cast<int>(sceneW / (kGridVLines + 1));
+    for (int i = 1; i <= kGridHLines; i++) {
+        m_pxfScene->addLine(QLineF(0, i * kGridHDist, sceneW, i * kGridHDist), gridPen);
     }
-    for (int i = 1; i < kGrindYLines; i++) {
-        m_pxfScene->addLine(
-                QLineF(i * (sizeX / kGrindYLines), 0, i * (sizeX / kGrindYLines), sizeY), gridPen);
+    for (int i = 1; i <= kGridVLines; i++) {
+        m_pxfScene->addLine(QLineF(i * kGridVDist, 0, i * kGridVDist, sceneH), gridPen);
     }
 
-    // Draw graph lines
-    QPointF pointTotal, point1, point2;
-    QPointF pointTotalPrev, point1Prev, point2Prev;
-    int pointCount = sizeX - 4;
-    // reduced by 2 x 1 for border + 2 x 1 for inner distance to border
-    double xfadeStep = 2. / (pointCount - 1);
-    for (int i = 0; i < pointCount; i++) {
-        CSAMPLE_GAIN gain1, gain2;
+    // Draw gain curves
+    QPolygonF polylineTotal;
+    QPolygonF polylineL;
+    QPolygonF polylineR;
+    // Required to have symmetrical curves not overdraw at the right
+    int pointCount = sceneW - 2;
+    double xfadeStep = 2. / (pointCount);
```

Comment:
what do these constants symbolize?

Reply:
* pixel steps for the graph calculation
* steps the xfader has to go to cover the CO (potmeter) range -1..0..1